node/src/proxy_server/mod.rs

```diff
@@ -162,12 +164,35 @@ impl AddReturnRouteMessage {
     }
 }
 
-impl Handler<AddRouteMessage> for ProxyServer {
+impl Handler<RouteResultMessage> for ProxyServer {
     type Result = ();
 
-    fn handle(&mut self, msg: AddRouteMessage, _ctx: &mut Self::Context) -> Self::Result {
-        debug!(self.logger, "Establishing stream key {}", msg.stream_key);
-        self.stream_key_routes.insert(msg.stream_key, msg.route);
+    fn handle(&mut self, msg: RouteResultMessage, _ctx: &mut Self::Context) -> Self::Result {
+        let dns_failure = self
+            .dns_failure_retries
+            .get(&msg.stream_key)
+            .unwrap_or_else(|| {
+                panic!("RouteResultMessage Handler: stream key: {} not found within dns_failure_retries", msg.stream_key);
```

Comment:
Are we absolutely certain sure that this can't be triggered by anything that comes in from outside over the network? If it can, we need to log a message here and continue, rather than panicking.

Reply:
Yes, we are certain that it can't be triggered by the outside network.

Here are the following reasons to justify this statement:

1) The `stream_key` is created as soon as the client generates a new request and the Node adds a new entry inside the HashMap as soon as that happens.
2) By the time this code executes we are certain that there must be an entry inside the HashMap for the desired `stream_key`, if that's not the case there must be some miscommunication between the client and the Node.
3) Also, the `RouteResultMessage` contains information from `Neighbourhood` that's intended for `ProxyServer`. Since, no information is dependent on the information triggered from outside the network, it's completely safe to conclude that this panic can't be triggered from outside the network.